File: backend/app/routes/project_normalization.py
from fastapi import APIRouter, HTTPException
from app.models.project_normalization_input import ProjectNormalizationInput
from app.agents.project_normalization_agent import ProjectNormalizationAgent

router = APIRouter(
    prefix="/api/project-normalization",
    tags=["project-normalization"],
)


@router.post("/build")
def build_normalized_context(inp: ProjectNormalizationInput):
    """
    Builds Normalized Intermediate Model (NIM)
    from Project Detection output.
    """
    try:
        agent = ProjectNormalizationAgent()
        out = agent.run(inp)
        return out
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Project normalization failed: {str(e)}",
        )

File: ui/ui_app/src/api/project_normalization.js
// src/api/projectNormalization.js

import axios from 'axios';

//const API_BASE_URL = 'http://127.0.0.1:5000';
const API_BASE_URL = 'http://localhost:8000';

/**
 * Calls Agent-2: Project Normalization (NIM) backend API
 * Input: Project Detection output
 * Output: Normalized Intermediate Model (NIM)
 */
export async function buildProjectNormalization(detectionResult) {
    try {
        const res = await axios.post(
            `${API_BASE_URL}/api/project-normalization/build`,
            detectionResult,
            {
                headers: {
                    'Content-Type': 'application/json',
                },
                timeout: 20000,
            }
        );

        return res.data;
    } catch (error) {
        const message =
            error?.response?.data?.detail ||
            error?.message ||
            'Unknown error during project normalization';

        return {
            error: true,
            message,
        };
    }
}